src/vs/editor/browser/widget/diffEditorWidget.ts

```diff
@@ -510,10 +506,13 @@ export class DiffEditorWidget extends Disposable implements editorBrowser.IDiffE
 			}
 		}));
 
+		this._register(Event.any(editor.onDidFocusEditorText, editor.onDidFocusEditorWidget)(() => contextKeyService.createKey('isInDiffLeftEditor', true)));
```

Comment:
I think the context key `editorFocus` also represents text or widget focus, I think you could bind the diff key on each side to that if you wanted.

Reply:
I tried to listen for the `editorFocus` context key, and I couldn't get it to work properly:

Here is what I tried, and it seemed that the change event wasn't fired with that key (which is strange)
```ts
const keys = new Set(['editorFocus']);
this._register(contextKeyService.onDidChangeContext(e => {
	if (e.affectsSome(keys)) {
		isInDiffRightEditorKey.set(contextKeyService.getContextKeyValue('editorFocus') ?? false);
	}
}));
```

At the same time, I think I like binding to the focus events better, since it is explicit and will fire far less often than the `contextKeyService.onDidChangeContext` which then needs to be filtered down.

Reply:
You will have to get the contextKeyService from inside the editor using `invokeWithinContext`

Reply:
But yeah the current code is fine.